mailchimp/mailchimp-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/mailChimp/MailChimpAuthenticator.java

```diff
@@ -0,0 +1,306 @@
+        if (log.isDebugEnabled()) {
+            log.debug("Claim URL: " + url);
+        }
+
+        if (url == null) {
+            return "";
+        } else {
+            URL obj = new URL(url);
+            HttpURLConnection urlConnection = (HttpURLConnection) obj.openConnection();
+            urlConnection.setRequestMethod("POST");
+            HttpClient httpClient = HttpClientBuilder.create().build();
+            HttpPost p = new HttpPost(url);
+
+            p.setEntity(new StringEntity("{\"apikey\":\"" + accessToken + "\"" + "}"));
+
+            HttpResponse r = httpClient.execute(p);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(r.getEntity().getContent()));
+            StringBuilder builder = new StringBuilder();
+
+            for (String inputLine = reader.readLine(); inputLine != null; inputLine = reader.readLine()) {
+                builder.append(inputLine).append("\n");
+            }
+
+            reader.close();
+            if (log.isDebugEnabled() && IdentityUtil.isTokenLoggable("UserIdToken")) {
+                log.debug("response: " + builder.toString());
+            }
+            return builder.toString();
+        }
+    }
+
+    private OAuthClientResponse getOauthResponse(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws AuthenticationFailedException {
+        OAuthClientResponse oAuthResponse = null;
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthSystemException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception while requesting access token", e);
+            }
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        } catch (OAuthProblemException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception while requesting access token", e);
+            }
+        }
+        return oAuthResponse;
+    }
+
+    private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientId, String code, String clientSecret,
+                                                String callbackurl) throws AuthenticationFailedException {
+        OAuthClientRequest accessRequest;
+        try {
+            accessRequest = OAuthClientRequest.tokenLocation(tokenEndPoint)
+                    .setGrantType(GrantType.AUTHORIZATION_CODE)
+                    .setClientId(clientId)
+                    .setClientSecret(clientSecret)
+                    .setCode(code)
+                    .setRedirectURI(callbackurl)
+                    .buildBodyMessage();
+        } catch (OAuthSystemException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception while building request for request access token", e);
+            }
+            throw new AuthenticationFailedException(e.getMessage(), e);
+        }
+        return accessRequest;
+    }
+
```

Comment:
Fixed in https://github.com/wso2-attic/is-connectors/pull/69/commits/b71eb18b4b2c1b5d353c06da8b56e2395196ba91